gui/default/syncthing/settings/settingsModalView.html

```diff
@@ -159,6 +159,153 @@
+                </div>
+              </div>
+              <div class="col-md-6">
+                <div class="form-group">
+                  <label translate for="WebauthnOrigin">WebAuthn Origin</label>
+                  <input id="WebauthnOrigin" class="form-control text-monospace" type="text" ng-model="tmpGUI.webauthnOrigin" placeholder="{{ getDefaultWebauthnOrigin(tmpGUI) }}" ng-disabled="!tmpGUI.useTLS" />
+                </div>
+              </div>
+            </div>
+            <div class="row">
+              <div class="col-md-12">
+                <p class="text-warning" ng-if="webauthnRpIdChanged(config.gui, tmpGUI) && tmpGUI.webauthnCredentials.length > 0" translate>
+                  Changing the RP ID will make existing credentials stop working.
+                </p>
+
+                <p class="text-warning" ng-if="!webauthnRpIdMatchesLocation(tmpGUI) && !isRawIpAddress()" translate translate-value-current-domain="{{ getLocationHost() }}">
+                  RP ID should be equal to or a parent domain of the current webpage domain: "{%currentDomain%}"
+                </p>
+
+                <p class="text-warning" ng-if="isRawIpAddress(tmpGUI.webauthnRpId || 'localhost')" translate>
+                  WebAuthn is not supported on raw IP address hosts.
+                </p>
+
+                <p ng-if="!locationMatchesWebauthnOrigin(tmpGUI) && !isRawIpAddress(tmpGUI.webauthnRpId || 'localhost')">
+                  <button ng-if="settingsModified()" type="button" class="btn btn-primary" ng-click="reloadAtWebauthnAddress(true, tmpGUI)">
+                    <span class="fas fa-check"></span>&nbsp;<span translate>Save settings and reload page at WebAuthn Origin</span>
+                  </button>
+
+                  <button ng-if="!settingsModified()" type="button" class="btn btn-primary" ng-click="reloadAtWebauthnAddress(false, tmpGUI)" translate>
+                    Reload page at WebAuthn Origin
+                  </button>
+                </p>
+              </div>
+            </div>
+
+            <!-- WebAuthn credentials -->
+            <div class="table-responsive webauthn-credentials">
+              <table class="table-striped table">
+                <thead>
+                  <tr>
+                    <th colspan="1000"><label translate>WebAuthn Credentials</label></th>
+                  </tr>
+                  <tr>
+                    <th><label translate="settings.webauthn.credentials.columnHeading.name">Name</label></th>
+                    <th><label translate="settings.webauthn.credentials.columnHeading.createTime">Created</label></th>
+                    <th><label translate="settings.webauthn.credentials.columnHeading.lastUseTime">Last used</label></th>
+                    <th><label translate="settings.webauthn.credentials.columnHeading.requireUserVerification">Require UV</label></th>
```

Comment:
By the way, I've used namespaced translation IDs here and in a few other places because the inferred ones don't work across languages and contexts. For example: in Swedish, "Created" could be translated as "Skapades" (_was_ created), "Skapad" (_is_ created, common) or "Skapat" (_is_ created, neuter). So a single "Created" translation key would fairly likely be translated incorrectly if the translator sees no other context, and almost certainly if it's used in more than one place.

Similarly, "Require UV" is difficult to translate at all without the context of what "UV" expands to, but it's also not appropriate to spell out "Require user verification" here since that would make the column heading far too wide.

Reply:
I don't think those are going to work. The current translation script only supports two types of syntax: `<tag translate>string</tag>`, and `{{string | translate}}`. We had a somewhat lengthy discussion on using other translation syntax types in https://github.com/syncthing/syncthing/pull/8548, however it ended without any conclusion other than that the translation script should always be able to regenerate `lang-en.json` automatically, i.e. no manual additions are allowed.

You can test the current translation script out by running `go run build.go translate` and see what actually happens.

Reply:
Indeed, if such syntax should be supported and is required here, we will need to extend the translation script. The mentioned discussion already had an implementation draft, but there should be a more general concept for what the translation keys should look like, if not always identical to the English string.

Reply:
> we will need to extend the translation script

[I did.](https://github.com/syncthing/syncthing/pull/9175/commits/a0b8407acbedd7b62d4f212083744973816275a2) :slightly_smiling_face: The screenshots above were made after running `go run build.go translate` with these additions.

Though I do notice now that if I switch to a different language without these keys translated, the UI displays the raw translation keys instead of falling back to English. Commit https://github.com/syncthing/syncthing/pull/9175/commits/1c5e4ba8adf13162f219a318f5012f3fe41a13fe should fix this too, though.

Reply:
I'd be happy to extract this discussion to a separate PR/issue if that seems appropriate.

Note that separating namespaces with `.` is a feature of angular-translate, which supports structured JSON:

```json
    "settings": {
        "webauthn": {
            "credentials": {
                "action": {
                    "add": "Add",
                    "delete": "Delete",
                    "rename": "Rename"
                },
                "columnHeading": {
                    "actions": "Actions",
                    "createTime": "Created",
                    "lastUseTime": "Last used",
                    "name": "Name",
                    "requireUserVerification": "Require UV"
                }
            }
        }
    },
```
(Which is also what `translate.go` generates with these additions.)

Reply:
Yes, please let's solve the translation issue first, decoupled from this PR. I don't see how the quoted JSON would work with the Weblate integration. Maybe there is support for it, if the structure is common practice among UI frameworks.

Reply:
Translation stuff extracted to PR #9192.